pkg/messaging/v1beta1/subscriptions_client.go

```diff
@@ -0,0 +1,189 @@
+	_, err := c.client.Update(subscription)
+	return knerrors.GetError(err)
+}
+
+// DeleteSubscription deletes Subscription by its name
+func (c *subscriptionsClient) DeleteSubscription(name string) error {
+	return knerrors.GetError(c.client.Delete(name, &metav1.DeleteOptions{}))
+}
+
+// ListSubscription lists subscriptions in configured namespace
+func (c *subscriptionsClient) ListSubscription() (*v1beta1.SubscriptionList, error) {
+	subscriptionList, err := c.client.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, knerrors.GetError(err)
+	}
+
+	return updateSubscriptionListGVK(subscriptionList)
+}
+
+func updateSubscriptionListGVK(subscriptionList *v1beta1.SubscriptionList) (*v1beta1.SubscriptionList, error) {
+	subscriptionListNew := subscriptionList.DeepCopy()
+	err := updateMessagingGVK(subscriptionListNew)
+	if err != nil {
+		return nil, err
+	}
+
+	subscriptionListNew.Items = make([]v1beta1.Subscription, len(subscriptionList.Items))
+	for idx, subscription := range subscriptionList.Items {
+		subscriptionClone := subscription.DeepCopy()
+		err := updateMessagingGVK(subscriptionClone)
+		if err != nil {
+			return nil, err
+		}
+		subscriptionListNew.Items[idx] = *subscriptionClone
+	}
+	return subscriptionListNew, nil
+}
+
+// SubscriptionBuilder is for building the Subscription object
+type SubscriptionBuilder struct {
+	subscription *v1beta1.Subscription
+}
+
+// NewSubscriptionBuilder for building Subscription object
+func NewSubscriptionBuilder(name string) *SubscriptionBuilder {
+	return &SubscriptionBuilder{subscription: &v1beta1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}}
+}
+
+// NewSubscriptionBuilderFromExisting for building Subscription object from existing Subscription object
+func NewSubscriptionBuilderFromExisting(subs *v1beta1.Subscription) *SubscriptionBuilder {
+	return &SubscriptionBuilder{subscription: subs.DeepCopy()}
+}
+
+// Channel sets the channel reference for this subscription
+func (s *SubscriptionBuilder) Channel(channel *corev1.ObjectReference) *SubscriptionBuilder {
+	if channel == nil {
+		return s
+	}
+
+	s.subscription.Spec.Channel = *channel
+	return s
+}
+
+func (s *SubscriptionBuilder) Subscriber(subs *duckv1.Destination) *SubscriptionBuilder {
+	if subs == nil {
+		return s
+	}
+
+	s.subscription.Spec.Subscriber = subs
+	return s
+}
+
+func (s *SubscriptionBuilder) Reply(reply *duckv1.Destination) *SubscriptionBuilder {
+	if reply == nil {
+		return s
+	}
+
+	s.subscription.Spec.Reply = reply
+	return s
+}
+
+func (s *SubscriptionBuilder) DeadLetterSink(dls *duckv1.Destination) *SubscriptionBuilder {
+	if dls == nil {
+		return s
+	}
+
+	ds := &eventingduckv1beta1.DeliverySpec{}
+	ds.DeadLetterSink = dls
+	s.subscription.Spec.Delivery = ds
+	return s
+}
+
+// Build returns the Subscription object from the builder
+func (c *SubscriptionBuilder) Build() *v1beta1.Subscription {
+	return c.subscription
```

Comment:
Just a cosmetic one.
```suggestion
func (s *SubscriptionBuilder) Build() *v1beta1.Subscription {
	return s.subscription
```